### DISCUSSION TITLE
FW 2.03 has connection problems

### DISCUSSION THREAD

**Discussion by heiniw** [simpleavr/s14clock#18] [General]
Hi, 

I received the clock with FW 2.01 installed. The clock worked fine but I wanted to upgrade it to the latest (and greatest) firmware. I flashed FW 2.03 onto it. It came back fine and I was able to reconfigure my Wifi. However, after that it didn't want to connect to my Wifi (showed 'Connecting...' on the display). After a power cycle, the clock came up and connected to Wifi.... only to lose it again after some minutes. Then it got back into the setup-mode and I had to connect again through the AP-mode. This happened a number of times in a row. 

I downgraded to FW 2.02. That version doesn't have the Wifi-problems and has been running now for some time without any problems.

FW 2.03 seems to have some bugs in it...

**Comment by simpleavr** [comment]
Thanks for your info. I will try the procedure in depth and on multiple devices to reproduce and eventually resolve the issue.

Please use FW2.0.2 for now. FW2.0.3 is rather new and I am still doing burn in testing and not annoucing it. FW2.0.3 only adds support for TZ string w/ daylight saving time.

I will post my findings and fix on this thread when they become available.

I am leaving FW2.0.3 binary and it is worthwhile to have others try FW2.0.3 and report if there are other instances of failures (or successes). The "firmware install" page had been fixed and should flash the S2Minis properly now.

**Comment by simpleavr** [comment]
I tried the FW2.03 via the install page and cannot reproduce the fail to connect (after setup) problem. I had tried 2 units and both works for me well.
I also examine the code and there were no changes regarding wifi credential / connection side.
I am leaving the firmware as if for now and see if there are additional reports of similar issue.

@heiniw If possible, please try to install FW2.03 again, sometimes in a congested LAN / far from router, the "connecting" can take a little while. Try move clock unit closer to router for stronger signal, or reset clock unit.

**Reply by heiniw** [reply]
As I have no connection issues with FW2.02, I don't think repositioning the module would make any difference. The problem with FW2.03 wasn't just that it took a little while; it would also give up and jump back into AP-initialisation mode.

I'm quite happy with FW2.02 right now.

Good luck on further development !
